src/uwtools/tests/test_rocoto.py

```diff
@@ -129,6 +129,44 @@ def test__add_task_dependency_fail(self, instance, root):
         with raises(UWConfigError):
             instance._add_task_dependency(e=root, config=config)
 
+    # pylint: disable=line-too-long
+    @pytest.mark.parametrize(
+        "config", [{"datadep": {"attrs": {"age": "120"}}}, {"timedep": {"attrs": {"offset": "&DEADLINE;"}}}]
+    )
+    def test__add_task_dependency_operand(self, config, instance, root):
+        instance._add_task_dependency_operand(e=root, config=config)
+        element = root[0]
+        for tag, attrs in config.items():
+            assert tag == element.tag
+            for attr, val in attrs["attrs"].items():
+                assert element.get(attr) == val
+
+    @pytest.mark.parametrize(
+        "config", [{"and": {"datadep": {"attrs": {"age": "120"}}, "timedep": {"attrs": {"offset": "&DEADLINE;"}}}}]
+    )
+    def test__add_task_dependency_operator(self, config, instance, root):
+        instance._add_task_dependency_operator(e=root, config=config)
+        element = root[0]
+        for tag, attrs in config.items():
+            assert tag == next(iter(config))
+            #for attr, val in attrs.items():
+            #    assert element.get(attr) == val
```

Comment:
Working on this final test case. Since the beginning key in the dict will not be `datadep`, `timedep`, or any other tags that will end the recursion, it seems that when it grabs `element = root[0]` it runs into a problem checking the rest of the `tags`/`attrs` later on. 

This gets even worse when nesting more operators ie..  `and` and `or` before we find one of these tags that stop the recursion. I think most of my issue is coming from not being able to properly manipulate the `element` variable in the debugger so I haven't been able to track down exactly what is happening.